examples/EResNN_CIFAR10.jl

```diff
@@ -1,78 +1,93 @@
-using MAT
-using Meganet
+using MAT, Meganet
 
-vars = matread("ECNN_compareCIFAR10.mat")
-Y = vars["Y0"]
-C = vars["C"]
-theta = vars["theta"]
-Z = vars["Z"]
-timeM = vars["timeMAT"]
+n = 64;
+Y_train,C_train,Y_test,C_test = getCIFAR10(n,string(pwd(),"//..//data//CIFAR10//"));
 
-miniBatchSize = 50;
+# using PyPlot
+# y = Y_train[:,50]; y = y - minimum(y); y = y./maximum(y);
+# y = reshape(y,32,32,3);
+# y[:,:,1] = y[:,:,1]';y[:,:,2] = y[:,:,2]';y[:,:,3] = y[:,:,3]';
+# figure(); imshow(y)
+
+miniBatchSize = 64;
 nImg = [32; 32]
 cin  = 3
 nc   = [16;32;64;64]
 nt   = 2*[1;1;1]
 h    = [1.;1.;1.]
 
+
+TYPE = Float32;
+
+
+# getConvKernel = (nImg,sK) -> getConvGEMMKernel(TYPE,nImg,sK);
+# getConvKernel = (nImg,sK) -> getConvFFTKernel(TYPE,nImg,sK);
+getConvKernel = (nImg,sK) -> getSparseConvKernel2D(TYPE,nImg,sK);
+
 # opening layer
-K1 = getSparseConvKernel2D(nImg,[3,3,cin,nc[1]])
-nL = getBatchNormLayer([prod(nImg);nc[1]],isTrainable=true)
-blocks = [singleLayer(K1,nL)]
+K1 = getConvKernel(nImg,[3,3,cin,nc[1]]);
+
+nL = getBatchNormLayer(TYPE,[prod(nImg);nc[1]],isTrainable=true);
+blocks = [getSingleLayer(TYPE,K1,nL)]
 
 for k=1:length(nt)
     # ResNN layers
-    K2 = getSparseConvKernel2D(nImg,[3,3,nc[k],nc[k]])
-    nL = getBatchNormLayer([prod(nImg);nc[k]],isTrainable=true)
-    L2 = DoubleSymLayer(K2,nL)
-    RN  = ResNN(L2,nt[k],h[k])
+    K2 = getConvKernel(nImg,[3,3,nc[k],nc[k]])
+    nL = getBatchNormLayer(TYPE,[prod(nImg);nc[k]],isTrainable=true)
+    L2 = getDoubleSymLayer(TYPE,K2,nL)
+    RN  = getResNN(TYPE,L2,nt[k],h[k])	
+	
     if k<length(nt)
         RN.outTimes *=0
     end
     blocks = [blocks;RN]
```

Comment:
`push!(blocks, RN)` and `append!` is often clearer, and for array types (returns true for` isbits()`) can often be done without creating a temporary array.